### ISSUE TITLE
cmd/go: should packages with paths that contain "/testdata/" be importable?

### ISSUE THREAD

**Issue by adg** [golang/go#60296]
The `cmd/go` docs [explicitly say](https://pkg.go.dev/cmd/go#hdr-Package_lists_and_patterns):

> Directory and file names that begin with "." or "_" are ignored by the go tool, as are directories named "testdata".

so it was very surprising to me to find that package [golang.org/x/crypto/ssh](https://pkg.go.dev/golang.org/x/crypto/ssh) has [an import](https://cs.opensource.google/go/x/crypto/+/master:ssh/testdata_test.go;l=15) of package `golang.org/x/crypto/ssh/testdata`, and that this passes tests!

<pre>
$ go list -f '{{range .TestImports}}{{printf "%s\n" .}}{{end}}' | grep testdata
golang.org/x/crypto/ssh/testdata
~/src/crypto/ssh $ go list -json ./testdata
{
	"Dir": "/Users/adg/src/crypto/ssh/testdata",
	"ImportPath": "golang.org/x/crypto/ssh/testdata",
	"Name": "testdata",
	"Doc": "This package contains test data shared between the various subpackages of the golang.org/x/crypto/ssh package.",
	"Root": "/Users/adg/src/crypto",
	"Module": {
		"Path": "golang.org/x/crypto",
		"Main": true,
		"Dir": "/Users/adg/src/crypto",
		"GoMod": "/Users/adg/src/crypto/go.mod",
		"GoVersion": "1.17"
	},
	"Match": [
		"./testdata"
	],
	"GoFiles": [
		"doc.go",
		"keys.go"
	]
}
</pre>

What's going on here? Did this sneak in by mistake? If so, we're stuck with it I suppose. Should the cmd/go docs be updated? What should the official rule be?

**Comment by ianlancetaylor** [comment]
CC @bcmills @matloob

**Comment by bcmills** [comment]
Yes: `testdata` packages should be excluded from patterns like `./...`, but should be importable.  (They have syntactically valid import paths, and it's convenient to, say, put a command in `testdata/cmd/foo` and be able to `go build` that command to use it in a test.)
